src/components/auth/Login.tsx

```diff
@@ -0,0 +1,87 @@
+import { createSupabaseClient } from "@/lib/supabase/client";
+import { useUser } from "@/contexts/UserContext";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+
+export function Login() {
+  try {
+    const { session, loading } = useUser();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+      // If user is already logged in, redirect to home
+      if (!loading && session?.access_token) {
+        navigate('/', { replace: true });
+      }
+    }, [session, loading, navigate]);
+
+    const handleLogin = async (provider: 'google' | 'github') => {
+      try {
+        const supabase = createSupabaseClient();
+        
+        const { error } = await supabase.auth.signInWithOAuth({
+          provider,
+          options: {
+            redirectTo: `${window.location.origin}/auth/callback`,
+            skipBrowserRedirect: false,
+            queryParams: {
+              access_type: 'offline',
+              prompt: 'consent',
+            }
+          },
+        });
+
+        if (error) {
+          console.error('[Login] Login error:', error.message);
+        }
+      } catch (error) {
+        console.error('[Login] Error creating Supabase client or signing in:', error);
```

Comment:
Try catch inside a react component is a bit odd, not usual

Reply:
Try catch on line 7 is weird. But this one is quite usual. I've used try catch within handler function.

Although hooks must not be wrapped in try catch. You can use try catch inside the hooks

It's an onclick handler function and it's completely fine. But Can use currying function to ensure component doesn't rerender

eg 
```
const handleLogin = useCallback((loginMethod: string) => () {

}, [])
```

and on the button 
```
onClick={handleLogin('google')}
```